### ISSUE TITLE
Evaluators ambiguity: class/module

### ISSUE THREAD

**Issue by r0mer0m** [embeddings-benchmark/mteb#1124]
There is ambiguity between classes and modules under the `evalutors` module. Specifically, evaluators objects under [mteb](https://github.com/embeddings-benchmark/mteb/tree/main/mteb)/[evaluation](https://github.com/embeddings-benchmark/mteb/tree/main/mteb/evaluation)/[evaluators](https://github.com/embeddings-benchmark/mteb/tree/main/mteb/evaluation/evaluators) are named the same as the modules which contain them. For example:

`mteb.evaluation.evaluators.RerankingEvaluator` references the class `RerankingEvaluator` in `mteb/evaluation/evaluators/RerankingEvaluator.py`. 

This is problematic because it does not allow to reference the `mteb/evaluation/evaluators/RerankingEvaluator.py` module and therefore referencing any other definition within it.

**Comment by KennethEnevoldsen** [comment]
I agree that this is a general problem within mteb. Ideally, we should have the modules in lowercase to avoid confusing the module with the object. I believe this is also the [pep8 recommendation](https://peps.python.org/pep-0008/#package-and-module-names).

We also have quite a high usage of `*` imports, which also leads to a few oddities e.g.:

```
from mteb import load_dataset
```

I would love a fix for this. Will just check in with @Muennighoff and @isaac-chung as well to make sure that everyone agrees

**Comment by isaac-chung** [comment]
Thanks for bringing this up! I think it's great if we can follow something like pep8. This would lead to a breaking change that might not be backwards compatible, so this would trigger a major version bump. We should consider this for perhaps for 2.0.

**Comment by Muennighoff** [comment]
> This is problematic because it does not allow to reference the mteb/evaluation/evaluators/RerankingEvaluator.py module and therefore referencing any other definition within it.

But there is no other definition within `RerankingEvaluator.py` besides `RerankingEvaluator`, no?

Note that we also name all task & abstask files in this style e.g. https://github.com/embeddings-benchmark/mteb/blob/main/mteb/tasks/Classification/ara/TweetEmotionClassification.py - I think if we change it for evaluators we should also change it there but still unsure if the change is needed. 🤔

**Comment by r0mer0m** [comment]
Thanks @KennethEnevoldsen and @isaac-chung, +1 that following pep8 standards is a good path forwards. 

I think that @Muennighoff brings a valid point and this is my bad as `RerankingEvaluator` might not be the best example. A more relevant example would be the `RetrievalEvaluator.py` module if we want to reference classes such as `DenseRetrievalExactSearch` within that module.

**Comment by Samoed** [comment]
I think it already fixed in `v2` branch and this can be closed? @KennethEnevoldsen

**Comment by KennethEnevoldsen** [comment]
isn't `STSEvaluator` still contained in `STSEvaluator.py`? (though we have fixed many of the import issues in v2)

**Comment by Samoed** [comment]
Yes, it is. But how do we want to rename it? Also, should we rename tasks that have the same filename as their class?

**Comment by KennethEnevoldsen** [comment]
If we want to be pep8 compatible (it could also solve some future import issues), but I see that it would require a lot of changes (and would be a v2 change, if at all)

**Comment by Samoed** [comment]
Yes, I think we should use snake_case for file names. However, I'm not sure if Git will properly track the changes after renaming. I believe this should be one of the final changes we make.

**Comment by isaac-chung** [comment]
For abstasks, do we want the naming to be in snake case like:
```diff
- AbsTaskBitextMining.py
+ bitext_mining.py
```

**Comment by KennethEnevoldsen** [comment]
Yep, @isaac-chung, @Samoed, and I also discussed renaming tasks from `AbstaskAnyClassfication` > `AbstaskClassfication` and for any domain-specific task `AbstaskSummarization` > `AbstaskTextSummarization`

**Comment by Samoed** [comment]
Do we want to keep `task` in names? We can name like `AbsClassfication`

**Comment by isaac-chung** [comment]
We can even just do `Classification`

**Comment by KennethEnevoldsen** [comment]
I think the way that task discovery is currently set up, it uses the `AbsTask*` prefix, which is not ideal, but that is why it is called `AbsTaskAggregate`. 

I don't think it is a biggie, though. I would probably keep it as is.

**Comment by isaac-chung** [comment]
The original issue with evaluator and abstask filenames have been addressed. Feel free to use a diff issue for other topics.
